ginac/inifcns_trig.cpp

```diff
@@ -176,6 +176,24 @@ static ex sin_eval(const ex & x)
                 const ex ExOverPi = x_red/Pi;
                 if (ExOverPi.info(info_flags::integer))
                         return _ex0;
+
+                // Reflection at Pi/2
+                else {
+                        if (is_exactly_a<numeric>(coef_pi)) {
+                                const numeric c = ex_to<numeric>(coef_pi);
+                                if (c.is_rational()) {
+                                        const numeric num = c.numer();
+                                        const numeric den = c.denom();
+                                        const numeric rm = num.mod(den);
+                                        
+                                        if (2*rm.to_int() > den.to_int()) {
```

Comment:
No. Numerics can be of type MPZ, MPQ, and PYOBJECT. All arithmetic operations on them are available, and because numerics are polymorphic you don't need to know which types they are most of the time.